fem/coefficient.hpp

```diff
@@ -534,6 +541,34 @@ class VectorRestrictedCoefficient : public VectorCoefficient
                      const IntegrationRule &ir);
 };
 
+/// VectorFunctionCoefficient defined on a subset of domain or boundary attributes, srw
+class VectorFunctionRestrictedCoefficient : public VectorCoefficient
+{
+private:
+   void (*TDFunction)(const Vector &, double, int, Vector &);
```

Comment:
Could this be a functor instead of a function pointer?

Reply:
We're currently following the convention being used by the other classes within this file that are performing the same type of operations. 

It might be a question then should the other classes that are doing thing within coefficient.hpp also be converted over to functors. If so would we want a series of base functors implemented in coefficient.hpp or some other file that we would derive our functors off of?

I know for our case we're only use this class to apply our boundary conditions. So, the performance hit taken from not having the function calls inlined shouldn't be too bad for our specific case. However, if the swap over to functors can provide a decent speed up for other people using functions from this file then it might be something to look into.